Modules/Loadable/Data/qSlicerDataModule.cxx

```diff
@@ -85,12 +86,19 @@ QStringList qSlicerDataModule::dependencies() const
 //-----------------------------------------------------------------------------
 void qSlicerDataModule::setup()
 {
+  Q_D(const qSlicerDataModule);
+
   this->Superclass::setup();
 
-  qSlicerAbstractCoreModule* camerasModule =
-    qSlicerCoreApplication::application()->moduleManager()->module("Cameras");
   vtkSlicerCamerasModuleLogic* camerasLogic =
-    vtkSlicerCamerasModuleLogic::SafeDownCast(camerasModule ? camerasModule->logic() : nullptr);
+    vtkSlicerCamerasModuleLogic::SafeDownCast(this->appLogic()->GetModuleLogic("Cameras"));
+  // NOTE: here we assume that camerasLogic with a nullptr value can be passed
+  // to the qSlicerSceneReader. Therefore we trigger a warning but don't return
+  // immediately.
+  if (!camerasLogic)
+    {
+    qCritical() << Q_FUNC_INFO << ": Cameras module is not found";
```

Comment:
I am wondering if the warning should be completely removed as no warning were displayed before.

Reply:
I'm not sure myself, @jcfr. I tried to be cautions on the event of passing nullptr to `qSlicerSceneReader` a few lines below. As for now, it does not seem to trigger any unnecessary warning, but on the other hand it seems ulikely that the Cameras module logic is not in place at that point. @lassoan, any opinions on this?

Reply:
I agree with @RafaelPalomar that it is better to be cautious and add the warning. If we find that there are false alarms (warnings are logged while starting up or shutting down the application) then we can refine the logic accordingly.